src/services/rfqm_service.ts

```diff
@@ -432,11 +436,90 @@ export class RfqmService {
     /**
      * Process an orderHash as an RfqmJob. Throws an error if job must be retried
      */
-    // tslint:disable: prefer-function-over-method
     public async processRfqmJobAsync(orderHash: string): Promise<void> {
-        logger.info({ orderHash }, 'finished processing job');
-        // tslint:disable-next-line: custom-no-magic-numbers
-        return Math.random() > 0.5 ? Promise.reject('failing') : Promise.resolve();
+        logger.info({ orderHash }, 'start processing job');
+
+        // Get job
+        const job = await this.dbUtils.findJobByOrderHashAsync(orderHash);
+        if (job === undefined) {
+            throw new NotFoundError(`job for orderHash ${orderHash} not found`);
+        }
+
+        // Basic validation
+        const { calldata, makerUri, order, fee } = job;
+        if (calldata === undefined) {
+            await this.dbUtils.updateRfqmJobAsync(orderHash, {
+                status: RfqmJobStatus.Failed,
+                statusReason: 'Missing Calldata',
+            });
+            return;
+        }
+
+        if (makerUri === undefined) {
+            await this.dbUtils.updateRfqmJobAsync(orderHash, {
+                status: RfqmJobStatus.Failed,
+                statusReason: 'Missing makerUri',
+            });
+            return;
+        }
+
+        if (order === null) {
+            await this.dbUtils.updateRfqmJobAsync(orderHash, {
+                status: RfqmJobStatus.Failed,
+                statusReason: 'Missing order on fee',
+            });
+            return;
+        }
+
+        if (fee === null) {
+            await this.dbUtils.updateRfqmJobAsync(orderHash, {
+                status: RfqmJobStatus.Failed,
+                statusReason: 'Missing fee on job',
+            });
+            return;
+        }
+
+        // Start processing
+        await this.dbUtils.updateRfqmJobAsync(orderHash, {
+            status: RfqmJobStatus.Processing,
+        });
+
+        // Validate w/Eth Call
+        try {
+            await this._blockchainUtils.decodeMetaTransactionCallDataAndValidateAsync(calldata, 'TODO Sender');
```

Comment:
this will be called by the worker, which will be aware of its own address. I think that can be an input to the function

Reply:
haha nice, you caught the unfinished portion of this diff - I think you're right, passing in the address as an input to this function is the best way to handle this (as opposed to initializing the rfqmService with an optional workerAddress)